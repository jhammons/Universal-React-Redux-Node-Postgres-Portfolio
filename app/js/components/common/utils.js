export function animation(animationArea, animationType="fadeInUp") {

    if (typeof window === 'undefined') return;

    const viewportHeight = Math.max(document.documentElement.clientHeight, window.innerHeight) || 0;

    return !window.location.hash && window.pageYOffset < viewportHeight * 0.8 ?
        (animationArea ? `animated ${animationType} ` : ` invisible `)
            : '';
}