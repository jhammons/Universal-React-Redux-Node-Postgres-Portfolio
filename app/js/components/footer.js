import React, {Component} from 'react';
import ContactForm from './contactForm'
import * as contactActions  from '../actions/contact';

export default class Footer extends React.Component {

    render() {
        const renderContactForm = this.props.contactForm.readyState === 'SUBMIT_CONTACT_SUCCESS' ?
            <p className="thanks">Thanks for your email!</p> :
            <ContactForm {...this.props} />;

        return (
            <footer className="footer">
                <div className="border"></div>
                <div className="container-fluid">
                    <div className="row" className="contact" id="contact">
                        <div className="col-lg-4">
                            <h4>CONTACT</h4>
                            <p className="contact-method"><i className="fa fa-phone"></i> (415) 287-3221</p>
                            <p className="contact-method"><i className="fa fa-envelope"></i> <a href="mailto:customerservice@jasongallagher.org?Subject=Interested%20in%20your%20services" target="_top">Email me</a></p>
                            <p className="contact-method skype"><i className="fa fa-skype"></i> <a href="skype:jason-gallagher?chat">Skype me</a></p>
                            <h4 className="about-this-site">ABOUT THE SITE</h4>
                            <p>This site is HANDCRAFTED with React, Redux, Node, Mysql, and OF COURSE, html and css. It is UNIVERSAL, or rather a single page app that is first rendered on the server.</p>
                        </div>
                        {renderContactForm}
                    </div>
                </div>
            </footer>
        );
    }
}