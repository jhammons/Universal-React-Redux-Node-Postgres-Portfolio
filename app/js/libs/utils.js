function getElemScrollPos(elem) {
    return document.getElementById(elem).getBoundingClientRect().top;
}

export function getEnvironment(type) {
    if (type === 'client') return typeof window !== 'undefined';
    if (type === 'production') return process.env.NODE_ENV === 'production';

    return {
        env: process.env.NODE_ENV === 'production' ? 'prod' : 'dev',
        loc: typeof window === 'undefined' ? 'server' : 'client'
    }
}

export function getFullWindowHeight() {
    if (getEnvironment().loc === 'server') return 0;

    var body = document.body,
        html = document.documentElement;

    return Math.max( body.scrollHeight, body.offsetHeight,
        html.clientHeight, html.scrollHeight, html.offsetHeight ) || 0;
}

export function animation(animationArea, animateOff, animationType='zoomInUp') {
    return !animateOff ?
        (animationArea ? `animated ${animationType} ` : ` invisible `)
            : '';
}