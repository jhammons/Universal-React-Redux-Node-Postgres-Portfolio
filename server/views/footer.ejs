    </div>
</div>
    <script src="/admin.js"></script>
    <script src="/bundle.min.js"></script>
</body>
</html>