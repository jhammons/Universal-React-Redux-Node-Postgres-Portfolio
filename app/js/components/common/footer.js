import React, {Component} from 'react';
import {connect} from 'react-redux';
import ContactForm from './contactForm';
import OpenChat from './hocs/openChat';
import ReactGA from'react-ga';
import * as contactActions  from '../../actions/contact';
import {animation} from '../../libs/utils';

class Footer extends React.Component {

    render() {
        const renderContactForm = this.props.contactForm.readyState === 'SUBMIT_CONTACT_SUCCESS' ?
            <p className="thanks">Thanks for your email!</p> :
            <ContactForm {...this.props} />;

        const footerID = this.props.slug ? 'footer-interior' : '';

        const analytics = (type) => {
            ReactGA.event({
                category: 'button click',
                action: type
            });
        };

        return (
            <footer className="footer" id={footerID}>
                <div className="footer-border"></div>
                <div className="footer-wrap" id="contact-area">
                    <div
                        className={animation(this.props.global.animateContact, this.props.global.animateOff) + "container-fluid"}>
                        <div className="contact row" id="contact">
                            <div className="col-lg-4">
                                <h4>CONTACT</h4>
                                <p className="contact-method"><i className="fa fa-phone"></i> &nbsp;(415) 287-3221</p>
                                <p className="contact-method chat">
                                    <i className="fa fa-cloud"></i> &nbsp;<OpenChat><a href="#">Live Chat</a></OpenChat>
                                </p>
                                <p className="contact-method"><i className="fa fa-envelope"></i> &nbsp;
                                    <a href="mailto:customerservice@jasongallagher.org?Subject=Interested%20in%20your%20services"
                                        target="_top" onClick={analytics.bind(this, 'email in footer')}>Email
                                    </a>
                                </p>
                                <h4 className="about-this-site">ABOUT THE SITE</h4>
                                <p className="about-this-site-text">This site is 100% <span>HANDCRAFTED</span> with
                                    React, Redux, Node, Mysql, and <span>OF COURSE</span>, Html and CSS. It is <span>UNIVERSAL</span>,
                                    or rather a single page app that is first rendered on the server. Not to be mistaken
                                    for yet another <span>HIPSTER BUZZWORD</span>, this is really great for performance,
                                    accessibility and SEO!</p>
                            </div>
                            {renderContactForm}
                        </div>
                    </div>
                </div>
            </footer>
        );
    }
}

function mapStateToProps(state) {
    return {
        contactForm: state.contactForm,
        global: state.global
    };
}
function mapDispatchToProps(dispatch) {
    return {
        dispatch: dispatch
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Footer);