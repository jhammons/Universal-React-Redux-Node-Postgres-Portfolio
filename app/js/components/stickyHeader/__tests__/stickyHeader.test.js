import StickyHeader from '../stickyHeader';
import { Footer } from '../../footer/footer';
import { ReactGA } from '../../../libs/utils';

describe('<StickyHeader />', () => {
    let wrapper;
    let props = {
        visible: false,
        openBurger: jest.fn(),
        remoteId: 'admin'
    };

    beforeEach(() => {
        wrapper = shallow(<StickyHeader {...props} />);
    });

    test('renders without crashing', () => {
        expect(wrapper.find('.sticky-header')).toHaveLength(1);
    });

    test('snapshot matches', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('displays \`on\` or \`off\` class depending on \`visible\` prop ', () => {
        expect(wrapper.find('.sticky-header.off')).toHaveLength(1);
        expect(wrapper.find('.sticky-header.on')).toHaveLength(0);
        wrapper.setProps({ visible: true });
        expect(wrapper.find('.sticky-header.off')).toHaveLength(0);
        expect(wrapper.find('.sticky-header.on')).toHaveLength(1);
    });

    test('analytics hooks are called', () => {
        jest.spyOn(StickyHeader.prototype, "analytics");
        const mockReactGA = jest.spyOn(ReactGA, "event")
            .mockImplementation(() => {});

        wrapper = shallow(<StickyHeader { ...props } />);

        const phone = wrapper.find('.phone');
        phone.simulate('click');
        expect(wrapper.instance().analytics).toHaveBeenCalledWith('phone number in header');

        const chatStatus = wrapper.find('.chat-status');
        chatStatus.simulate('click');
        expect(wrapper.instance().analytics).toHaveBeenCalledWith('chat in header');

        expect(mockReactGA).toHaveBeenCalledTimes(2);
    });
});
