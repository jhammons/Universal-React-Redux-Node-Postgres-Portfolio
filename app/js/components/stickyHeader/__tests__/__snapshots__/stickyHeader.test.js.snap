// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StickyHeader /> snapshot matches 1`] = `
<div
  className="sticky-header hidden-xs-down off"
>
  <div
    className="container-fluid"
  >
    <Navigation
      home={true}
      logo="1"
      ulClass="sticky-nav"
    />
    <div
      className="hamburger-alt hidden-md-up pull-right"
      onClick={[MockFunction]}
    >
      <i
        className="fa fa-bars"
      />
       Â MENU
    </div>
    <div
      className="contact-info pull-right hidden-xs-down"
    >
      <div>
        <span
          className="phone"
          onClick={[Function]}
        >
          <i
            className="fa fa-phone"
          />
           (707) 280-3629
        </span>
        <span
          className="chat-status"
          onClick={[Function]}
        >
          <Connect(OpenChat)>
            <liveChatStatus
              remoteId="admin"
            />
          </Connect(OpenChat)>
        </span>
      </div>
    </div>
  </div>
</div>
`;
